skstats/hypotests/core/discovery.py

```diff
@@ -0,0 +1,67 @@
+from scipy.stats import norm
+
+from .basetest import BaseTest
+
+
+class Discovery(BaseTest):
+    def __init__(self, calculator, poinull):
+        """Class for discovery test.
+
+            Args:
+                calculator (`sktats.hypotests.BaseCalculator`): calculator to use for computing the pvalues
+                poinull (List[`hypotests.POI`]): parameters of interest for the null hypothesis
+
+            Example with `zfit`:
+                >>> import zfit
+                >>> from zfit.core.loss import ExtendedUnbinnedNLL
+                >>> from zfit.minimize import MinuitMinimizer
+
+                >>> bounds = (0.1, 3.0)
+                >>> zfit.Space('x', limits=bounds)
+
+                >>> bkg = np.random.exponential(0.5, 300)
+                >>> peak = np.random.normal(1.2, 0.1, 25)
+                >>> data = np.concatenate((bkg, peak))
+                >>> data = data[(data > bounds[0]) & (data < bounds[1])]
+                >>> N = data.size
+                >>> data = zfit.data.Data.from_numpy(obs=obs, array=data)
+
+                >>> lambda_ = zfit.Parameter("lambda", -2.0, -4.0, -1.0)
+                >>> Nsig = zfit.Parameter("Ns", 20., -20., N)
+                >>> Nbkg = zfit.Parameter("Nbkg", N, 0., N*1.1)
+                >>> signal = Nsig * zfit.pdf.Gauss(obs=obs, mu=1.2, sigma=0.1)
+                >>> background = Nbkg * zfit.pdf.Exponential(obs=obs, lambda_=lambda_)
+                >>> loss = ExtendedUnbinnedNLL(model=[signal + background], data=[data], fit_range=[obs])
+
+                >>> from skstats.hypotests.calculators import AsymptoticCalculator
+                >>> from skstats.hypotests import Discovery
+                >>> from skstats.hypotests.parameters import POI
+
+                >>> calculator = AsymptoticCalculator(loss, MinuitMinimizer())
+                >>> poinull = POI(Nsig, 0)
+                >>> discovery_test = Discovery(calculator, [poinull])
+                >>> discovery_test.result()
```

Comment:
I think this will fail doctests as there is no corresponding printout. Get back to it once doctests are running in the CI … (Sorry, I did not yet check whether they are running already :D.)

Reply:
Ah! I try and have examples for anything that is not trivial, and then it really is important to have doctests running, to lower the burden of maintenance as much as possible. Indeed it is very easy to change something and quickly have examples in the code that no longer work :S. Having the doctests in `Particle` and `DecayLanguage` has been super useful IMHO.